nitime/timeseries.py

```diff
@@ -309,6 +316,11 @@ def mean(self, *args, **kwargs):
         return ret
 
     def ptp(self, *args, **kwargs):
+        if _NP_2:
+            raise AttributeError(
+                "`ptp` was removed from the ndarray class in NumPy 2.0. "
+                "Use np.ptp(arr, ...) instead."
+            )
         ret = TimeArray(np.ndarray.ptp(self, *args, **kwargs),
```

Comment:
The goal here seems to be to return a `TimeArray`, not an `ndarray`. I would probably just do:

```suggestion
            ptp = np.ptp
        else:
            ptp = np.ndarray.ptp
        ret = TimeArray(ptp(self, *args, **kwargs),
```

This will avoid breakage in the short term. If somebody wants `np.ptp(timearray, ...)` to work, I think the better path forward would be to refactor `TimeArray` to use the Array API.

Either way, you're proposing an API change in a fixup PR, which seems bad.

Reply:
> Either way, you're proposing an API change in a fixup PR, which seems bad.

Well in the sense that it keeps `TimeArray` behaving like `np.ndarray` I think it's at least reasonable. It's `nitime` following NumPy's breakage of the method, when the breaking NumPy is installed. In this sense at least things are kept consistent, whereas an inconsistency with NumPy is introduced if we keep `ptp` working for this subclass when it doesn't for `ndarray` anymore.

> I would probably just do:...

I think I did this but I'll push the commit and maybe CIs will tell us why it does or doesn't work. I think there was a recursion issue because `TimeArray` implements the `ptp` method I think `np.ptp` tries to use it, which calls this function again, etc. One workaround would be to do
```
ret = TimeArray(np.ptp(np.array(self), *args, **kwargs),
```
i.e., make a copy of the array and force it to use `ptp` directly.

Reply:
What about `ret = TimeArray(np.ptp(self.view(np.ndarray), *args, **kwargs))`?

Reply:
What you had originally worked! I must have done something wrong when I tried something similar earlier.